Первичная реализация сервиса Товары

KiryakovDenis opened the pull request.

KiryakovDenis committed:
Первичная реализация сервиса Товары

gmnpr commented on src/main/java/ru/kdv/study/controller/ProductController.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @PostMapping("/exist")
+    @Operation(summary = "Проверить существова")
```
проверить существо :D

gmnpr commented on src/main/java/ru/kdv/study/repository/ProductRepository.java:
```diff
@@ -0,0 +1,108 @@
+
+    public Boolean existById(final Long id) {
+        try {
+            return jdbcTemplate.queryForObject(EXIST_BY_ID, new MapSqlParameterSource("id", id) , existProductMapper);
```
зря я писал комменты к сервису x6-user...

gmnpr commented on src/main/java/ru/kdv/study/service/ProductService.java:
```diff
@@ -0,0 +1,61 @@
+    public List<ProductExist> productExistList(List<Long> productList) {
+        return productList.stream()
+                .map((Long id) -> {
+                    return new ProductExist(id, productRepository.existById(id));
```
надо сделать все возможно, чтобы избежать обращений к базе из цикла

KiryakovDenis committed:
Ревью
Избавился от вызовов к бд в цикле, будь оно проклято
С грустью удалил веселое описание "Проверить существова"

KiryakovDenis committed:
Удалил файлы среды из репозитория

KiryakovDenis committed:
Чутка подправил форматирование + static -> static final

KiryakovDenis committed:
Добавил кэширование Redis

KiryakovDenis committed:
Удалил неиспользуемый импорт

KiryakovDenis committed:
Удалил из репозитория файлы фрэймворка Gradle

KiryakovDenis committed:
Допилил gitignore и добавил интерфейс Serializable к модели Product

KiryakovDenis committed:
Удалил свой адрес из настроек Swagger,
Исправил лямбды

KiryakovDenis committed:
Убраны лишние файлы gradlew

KiryakovDenis committed:
изменен .gitignore

gmnpr commented on src/main/java/ru/kdv/study/Exception/BadRequestException.java:
```diff
@@ -0,0 +1,12 @@
+package ru.kdv.study.Exception;
```
пакеты с маленькой буквы

gmnpr commented on src/main/java/ru/kdv/study/model/ServiceException.java:
```diff
@@ -0,0 +1,10 @@
+
+@AllArgsConstructor
+@Getter
+public class ServiceException {
```
наверное по аналогии с другими сервисами надо сделать ServiceError

KiryakovDenis committed:
Изменено название класса ServiceException -> ServiceExceptionMessage
Доработан codestyle
